image-diff

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+
+from machine import testvm
+
+
+def get_packages(machine):
+    if any(machine.image.startswith(p) for p in ['debian', 'ubuntu']):
+        pkgcmd = "dpkg-query -W -f='PACKAGE\t${source:Package}\t${version}\n'"
+    elif any(machine.image.startswith(p) for p in ['rhel', 'fedora', 'centos']):
```

Comment:
That still won't catch images like "openshift" or "services". I really suggest to detect this dynamically; as a first suggestion:
```sh
if dpkg -s bash >/dev/null  2>&1; then
    ... it's dpkg;
elif rpm -q bash >/dev/null 2>&1; then
   ... it's rpm;
else
   ... unknown package system, error out;
fi
```

Reply:
naja.  i was actually thinking to run both commands on every host.  the list of rpm packages installed on a debian system is gonna be pretty small :)

Reply:
That's a great idea as well. One *can* install packages from a foreign package manager, but (1) it's not going to be the case for our CI images, and (2) *if* it is, then it's actually correct to include them. You just need to account for one or the other executable not being present at all, but at the same time we shouldn't ignore errors if the dpkg/rpm command fails.

Reply:
I went with a fairly reasonable approach for now:  `dpkg 2>/dev/null || rpm 2>/dev/null`.  Will give us the output, either way, and will fail if neither is installed.  If both happen to be installed, then we'll have a problem ya...